NEWS.md

```diff
@@ -16,6 +16,7 @@
 
 ## Bug fixes
 
+* `expect_identical_linter()` no longer lints `expect_equal()` comparison to negative non-integers (#2411, @Bisaloo).
```

Comment:
I think this description requires a lot of context to understand, I would mention this is for parity with positive non-integers, maybe:

```suggestion
* `expect_identical_linter()` also skips `expect_equal()` comparison to _negative_ non-integers like `-1.034` (#2411, @Bisaloo). This is a parity fix since _positive_ reals have always been skipped because "high-precision" comparisons are typically done to get tests within `tolerance`, so `expect_identical()` is not a great substitution.
```